lib/bootloader_setup.pm

```diff
@@ -527,21 +527,14 @@ sub specific_caasp_params {
 
     # Wait for supportserver (controller node)
     if (!check_var 'STACK_ROLE', 'controller') {
-        mutex_lock 'dhcp';
-        mutex_unlock 'dhcp';
+        pause_until 'support_server_ready';
```

Comment:
JFYI, a function `wait_supportserver` already exists for this. But you could keep your `pause_until` here if you want to use the same function everywhere.